cmd/backfill-redis/main.go

```diff
@@ -106,45 +111,57 @@ func main() {
 		log.Fatalf("creating rekor client: %v", err)
 	}
 
+	ctx, _ := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
+	group, ctx := errgroup.WithContext(ctx)
+	group.SetLimit(*concurrency)
+
 	for i := *startIndex; i <= *endIndex; i++ {
-		params := entries.NewGetLogEntryByIndexParamsWithContext(context.Background())
-		params.SetLogIndex(int64(i))
-		resp, err := rekorClient.Entries.GetLogEntryByIndex(params)
-		if err != nil {
-			log.Fatalf("retrieving log uuid by index: %v", err)
-		}
-		var insertErrs []error
-		for uuid, entry := range resp.Payload {
-			// uuid is the global UUID - tree ID and entry UUID
-			e, _, _, err := unmarshalEntryImpl(entry.Body.(string))
-			if err != nil {
-				insertErrs = append(insertErrs, fmt.Errorf("error unmarshalling entry for %s: %v", uuid, err))
-				continue
-			}
-			keys, err := e.IndexKeys()
+		index := i // capture loop variable for closure
+		group.Go(func() error {
+			params := entries.NewGetLogEntryByIndexParamsWithContext(ctx)
+			params.SetLogIndex(int64(index))
+			resp, err := rekorClient.Entries.GetLogEntryByIndex(params)
 			if err != nil {
-				insertErrs = append(insertErrs, fmt.Errorf("error building index keys for %s: %v", uuid, err))
-				continue
+				log.Fatalf("retrieving log uuid by index: %v", err)
```

Comment:
Can we change this to just print the error?

Reply:
We can, but I want to highlight that this is the current behavior (I wish the GitHub diff showed character-level diff so we can see that this whole block was indented and barely modified), and when I ran this on production (over ~50,000 entries), I never hit this. So IMO, I would rather retain this behavior -- bail if the Rekor call fails, since that likely indicates a config/networking problem. Let me know, I'm happy to change it if you disagree.

Reply:
Fair point, I'm ok with keeping this as is.